packages/server-wallet/src/models/objective.ts

```diff
@@ -156,12 +154,12 @@ export class ObjectiveModel extends Model {
     };
   }
 
-  static async insert<O extends DBObjective = DBObjective>(
-    objectiveToBeStored: SupportedObjective, // TODO this type should be correlated to O
+  static async insert<O extends SupportedObjective = SupportedObjective>(
+    objectiveToBeStored: O,
     preApproved: boolean,
     tx: TransactionOrKnex,
-    waitingFor?: WaitingFor // TODO this type should be correlated to O
-  ): Promise<O> {
+    waitingFor?: WaitingFor // TODO this should be correlated to O
```

Comment:
Could we take care of the TODO?

Reply:
We definitely should, but I think it is out of scope for this PR. I will make an issue for it.